src/sentry/integrations/gitlab/repository.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import
+
+import six
+
+from sentry.plugins import providers
+from sentry.models import Integration
+
+
+class GitlabRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'Gitlab'
+
+    def get_installation(self, integration_id, organization_id):
+        if integration_id is None:
+            raise ValueError('%s requires an integration_id' % self.name)
+
+        try:
+            integration_model = Integration.objects.get(id=integration_id)
+        except Integration.DoesNotExist as error:
+            self.handle_api_error(error)
+
+        return integration_model.get_installation(organization_id)
```

Comment:
Actually, maybe this part will fail if the `Integration` isn't associated with the `Organization`?

Reply:
I was wondering this because of my previous comment. If this fails when they're not associated, the previous comment about adding `organization` to the filter is less important.